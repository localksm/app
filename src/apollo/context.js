import React, {createContext} from 'react';

const Context = createContext({});

export const ContextProvider = Context.Provider;
export const ContextConsumer = Context.Consumer;

export const withContext = Comment => props => (
    <ContextConsumer>
        {state => <Component {...props} state={state} />}
    </ContextConsumer>
)
