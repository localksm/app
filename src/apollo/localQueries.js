import {gql} from 'apollo-boost';

export const SESSION = gql`
  query GetSessionValue {
    session @client {
      token
      email
      name
      balance
      id
      photo
      sessionType
    }
  }
`;

export const OFFER = gql`
  query GetOfferValue {
    offer @client {
      id
      status
      hash
      localCurrency
      body {
        paymentData {
          country
          email
          userId
          proposalId
          name
          lastName
          address
          phone
          bankData
          accountNumber
          type
        }
        makerId
        takerId
        makerType
        takerType
        operationType
        requestId
        requestAsset
        offerAsset
        requestAmount
        offerAmount
        timestamp
        paymentMethod
        usernameMaker
        usernameTaker
        toEmail
        adjudicationId
      }
    }
  }
`;

export const GET_FILTER_RESULT = gql`
  query GetFilterData {
    filterData @client {
      rangeLow
      rangeHigh
      order
    }
  }
`;

export const APOLLO_NAVIGATION = gql`
  query GetNavigation {
    navigation @client {
      navigate
    }
  }
`;

export const SELECTED_IMAGES = gql`
  query GetImges {
    evidenceImages @client {
      images
    }
  }
`;

export const TOKEN = gql`
  query GetToken {
    token @client {
      jwt
    }
  }
`;

export const GET_BALANCE_KSM = gql`
  query GetBalanceKSM {
    polkadot @client {
      balanceKSM
    }
  }
`;