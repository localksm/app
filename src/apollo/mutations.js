import {gql} from 'apollo-boost';

export const LOGIN_FACEBOOK = gql`
  mutation login(
    $email: String!,  
    $type:  SessionType!,
    $token: String,
    $userFBID: String,
    $platform: Platform
  ){
    login( email: $email, type: $type, token: $token, userFBID: $userFBID, platform: $platform  ){
      id
      email
      token
      success
      error
    }
  }`;

export const SIGNUP_FACEBOOK = gql `
  mutation signup(
    $name: String!,
    $email: String!,  
    $type:  SessionType!,
    $token: String,
    $userFBID: String,
    $platform: Platform
  ){
    signup( name: $name, email: $email, type: $type, token: $token, userFBID: $userFBID, platform: $platform  ){      
      success
      error
      userId
    }
  }
`;