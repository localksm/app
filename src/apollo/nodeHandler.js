function getNodeType(operationName) {
  switch (operationName) {
    case 'addFunds':
      return 'maker-seller';

    case 'withdrawFunds':
      return 'maker-buyer';

    default:
      return null;
  }
}

export default getNodeType;
