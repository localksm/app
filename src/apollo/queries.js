import { gql } from 'apollo-boost';

export const QUERY_EMAIL = gql`
  query verifyUser($email: String, $name: String) {
    verifyUser(email: $email, name: $name) {
      emailExists
      nameExists
    }
  }
`;

export const PAYMENT_METHODS = gql`
  query {
    paymentMethods {
      id
      value
    }
  }
`;

export const CURRENCIES = gql`
  query {
    currencies {
      id
      label
    }
  }
`;

export const COUNTRIES = gql`
  query {
    countries {
      id
      value
    }
  }
`;

export const PUBLIC_KEY = gql`
  query publicKeys($id: Int!) {
    publicKeys(id:$id){
      ksm
    }
  }
`;

export const QUERY_PROPOSALS = gql`
  query porposals($userId: Int, $offset: Int = 0, $limit: Int = 10) {
    proposals(
      userId: $userId
      offset: $offset
      limit: $limit
    ) {
      id
      status
      body {
        usernameMaker
        usernameTaker
        updatedAt
        requestId
        operationType
        makerId
        takerId
        offerAsset
        offerAmount
        requestAsset
        requestAmount
        paymentMethod
        paymentData {
          proposalId
          name
          lastName
          email
          address
          phone
          bankData
          accountNumber
          type
          country
        }
      }
    }
  }`;


export const QUERY_USER_PROPOSALS = gql`
query userProposals($id: Int, $limit: Int = 100,$offset: Int = 0 ) {
{  
    userProposals(id:$id limit:$limit offset:$offset ){ 
    id
        hash
        status
        count
        localCurrency
        body {
            paymentData {
                country
                email
                userId
                proposalId
                name
                lastName
                address
                phone
                bankData
                accountNumber
                type
            }
            requestId
            makerId
            takerId
            makerType
            takerType
            operationType
            offerAsset
            requestAsset
            offerAmount
            requestAmount
            timestamp
            conditions
            audience
            challengeStake
            juryPool
            paymentMethod
            usernameMaker
            usernameTaker
            from
            to
            toEmail
            remittanceSuccess
            fromUsername
            toUsername
            updatedAt
            adjudicationId
        }
    }
    }
}
`;
