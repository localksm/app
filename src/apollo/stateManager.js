import React from 'react';
import { client } from '.';


class ApolloState {
  constructor() {
    this.client = client;
  }

  mutation = async (payload, type = null) => {
    // Defautl saves
    this.cache.writeData({
      data: payload,
    });
  };
}

export default ApolloState;