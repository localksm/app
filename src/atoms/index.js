export { default as Button } from './Button';
export { default as InputText } from './InputText';
export { default as Link } from './Link';
export { default as Dropdown } from './Dropdown';
export { default as FBLoginButton } from './facebookButton';
export { default as GoogleButton } from './GoogleButton';
export { default as ButtonSignUp } from './ButtonSignUp';
export { default as Fees } from './Fees';
export { default as ButtonSignIn } from './ButtonSignIn';
export { default as DropdownPaymentMethods } from './DropdownPaymentMethods';
export { default as DropdownCountries } from './DropdownCountries';
export { default as DropdownCurrencies } from './DropdownCurrencies';
export { default as AddFundsButton } from './AddFundsButton';
export { default as QrWallet } from './QrCode';
export { default as ProposalTypeImages } from './ProposalTypeImages';
export { default as ConfirmSentBuyButton } from './ConfirmSentBuyButton';
export { default as ConfirmReceivedButton } from './ConfirmReceivedButton';
export { default as TWLoginButton } from './TwitterButton';
export { default as SellButton } from './SellButton';
export { default as InputLayout } from './InputLayout';
export { default as SendWithdrawButton } from './SendWithdrawButton';
export { default as SendAdjudicationButton } from './SendAdjudicationButton';
export { default as SendDisbursementBuyer } from './SendDisbursementBuyer';
export { default as SendDisbursementSeller } from './SendDisbursementSeller';
export { default as TextError } from './TextError';
export { default as OnboardingDoneButton } from './OnboardingDoneButton';
export { default as OnboardingNextButton } from './OnboardingNextButton';
export { default as OnboardingSkipButton } from './OnboardingSkipButton';
export { default as FeeContent} from './FeeContent';
export { default as Balance } from './Balance';
export { default as useSignIn } from './useSignIn';
export { default as UseGoogleButton } from './useGoogleButton';
export { default as UseSendWithdrawButton} from './useSendWithdrawButton';
export { default as UseBalance } from './useBalance';
export { default as useSellButton } from './useSellButton';
export { default as useButtonSignUp } from './useButtonSignUp';
export { default as useTwitterButton } from './useTwitterButton';
export { default as useFacebookButtom } from './useFacebookButtom';
export { default as useSendDisbursementBuyer } from './useSendDisbursementBuyer';
export { default as useSendDisbursementSeller } from './useSendDisbursementSeller';