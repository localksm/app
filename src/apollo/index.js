import {
  ApolloClient,
  ApolloLink,
  HttpLink,
  InMemoryCache,
} from 'apollo-boost';
import { WebSocketLink } from 'apollo-link-ws';
import { getMainDefinition } from 'apollo-utilities';
import { onError } from 'apollo-link-error';
import { setContext } from 'apollo-link-context';
import { split } from 'apollo-link';
import ls from 'react-native-local-storage';
import {
  GRAPHQL_ENDPOINT,
  GRAPHQL_SUBSCRIPTIONS_ENDPOINT,
} from '../utils/config';
import { ContextProvider, ContextConsumer, withContext } from './context';
import ApolloState from './stateManager';
import {
  SIGNUP,
  LOGIN,
  ADD_FUNDS,
  CONFIRM_PROPOSAL,
  SEND_ACCEPTANCE,
  SEND_RESOLUTION,
  SEND_SETTLEMENT,
  SEND_FULFILLMENT,
  INSERT_PROPOSAL_PAYMENT_METHOD,
  SEND_DISBURSEMENT_BUYER,
  SEND_DISBURSEMENT_SELLER,
  SELL,
} from './mutations';
import {
  VERIFY_USER,
  FEE,
  PAYMENT_METHODS,
  CURRENCIES,
  COUNTRIES,
  PUBLIC_KEY,
  QUERY_PROPOSALS,
  QUERY_USER_PROPOSALS,
} from './queries';

const httpLink = new HttpLink({
  uri: `${GRAPHQL_ENDPOINT}`,
});

const wsLink = new WebSocketLink({
  uri: `${GRAPHQL_SUBSCRIPTIONS_ENDPOINT}`,
  options: { reconnect: true },
});

export const setSession = async payload => {
  return await ls.save('session', payload);
};

export const getSession = async () => {
  return await ls.get('session');
};

export const removeSession = async () => {
  return await ls.clear();
};

const authLink = setContext(async _ => {
  const data = await getSession();
  const { session } = data !== null && data;
  const jwt = session && session.token;

  return {
    headers: {
      authorization: `Bearer ${jwt}`,
    },
  };
});

const httpAuthLink = authLink.concat(httpLink);

const errorLink = onError(({ graphQLErrors, networkError }) => {
  if (graphQLErrors) {
    graphQLErrors.map(({ message, locations, path }) =>
      // eslint-disable-next-line no-console
      console.log(
        `[GraphQL error]: Message: ${message}, Location: ${JSON.stringify(locations)}, Path: ${path}`,
      ),
    );
  }
  // eslint-disable-next-line no-console
  if (networkError) console.log(`[Network error]: ${networkError}`);
});

const link = split(
  ({ query }) => {
    const definition = getMainDefinition(query);
    return (
      definition.kind === 'OperationDefinition' &&
      definition.operation === 'subscription'
    );
  },
  wsLink,
  httpAuthLink,
);
export const cache = new InMemoryCache();
export const client = new ApolloClient({
  link: ApolloLink.from([errorLink, link]),
  cache,
  resolvers: {
    logout: async id => {
      await removeSession();
    },
  },
});

export const QUERIES = {
  VERIFY_USER,
  FEE,
  PAYMENT_METHODS,
  COUNTRIES,
  CURRENCIES,
  PUBLIC_KEY,
  QUERY_PROPOSALS,
  QUERY_USER_PROPOSALS,
};

export const MUTATIONS = {
  SIGNUP,
  LOGIN,
  SEND_ACCEPTANCE,
  SEND_RESOLUTION,
  SEND_SETTLEMENT,
  SEND_FULFILLMENT,
  INSERT_PROPOSAL_PAYMENT_METHOD,
  ADD_FUNDS,
  CONFIRM_PROPOSAL,
  SEND_DISBURSEMENT_BUYER,
  SEND_DISBURSEMENT_SELLER,
  SELL,
};

export const SUBSCRIPTIONS = {};

export { ContextConsumer, ContextProvider, withContext };

const initialState = {
  session: null,
  offer: null,
  filterData: [],
  navigation: null,
  evidenceImages: null,
  token: null,
};
cache.writeData({ data: initialState });

export const state = ApolloState;
