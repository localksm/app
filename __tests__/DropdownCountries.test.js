/**
 * @format
 */

import 'react-native';
import React from 'react';
import { shallow, configure } from 'enzyme';
import renderer, { act } from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { DropdownCountries } from '../src/atoms';
import { MockedProvider } from '@apollo/react-testing';
import { QUERIES } from '../src/apollo';
import { GraphQLError } from 'graphql';

configure({ adapter: new Adapter() });

const mocks = [
  {
    request: {
      query: QUERIES.COUNTRIES,
      variables: { },      
    },
    result: {
      data: {
        countries: [
          {
            id: 1,
            value: 'Mexico'
          }
        ],
      },
    },
  },
];

const mockErrors = [
  {
    request: {
      query: QUERIES.COUNTRIES,
      variables: { }
    },
    result: {
      errors: [],
    }
  }
];

it('renders correctly', async () => {
  const actionfake = jest.fn();
  await act(async () => {
    renderer.create(<DropdownCountries action={actionfake} />);
  });
});

it('should render loading state initially', async () => {
  await act(async () => {
    jest.useFakeTimers();
    const actionfake = jest.fn();
    const component = renderer.create(
      <MockedProvider mocks={[]}>
        <DropdownCountries action={actionfake} />
      </MockedProvider>,
    );
  });
});

it('mocks apollo data', async () => {
  await act(async () => {
    jest.useFakeTimers();
    const actionfake = jest.fn();
    const wrapper = renderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <DropdownCountries action={actionfake} />
      </MockedProvider>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});

it('mocks apollo error', async () => {
  await act(async () => {
    jest.useFakeTimers();
    const actionfake = jest.fn();
    const wrapper = renderer.create(
      <MockedProvider mocks={mockErrors} addTypename={false}>
        <DropdownCountries action={actionfake} />
      </MockedProvider>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
