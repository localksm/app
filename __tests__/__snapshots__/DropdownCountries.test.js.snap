// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mocks apollo data 1`] = `null`;

exports[`mocks apollo error 1`] = `null`;
